docs/administration/saml-integration.md

```diff
@@ -0,0 +1,141 @@
+
+This chapter covers the following subjects:
+
+-  [eXo Platform as SAML2 SP](#exo-platform-as-saml2-sp)
+
+-  [Generating and using your own keystore](#generating-and-using-your-own-keystore)
+
+## eXo Platform as SAML2 SP
+
+1. Install SAML2 add-on with the command:
+
+```bash
+$PLATFORM_SP/addon install exo-saml
+```
+
+2. Open the file `$PLATFORM_SP/gatein/conf/exo.properties`.
+
+Edit the following properties (add them if they don't exist):
+
+```properties
+# SSO
+gatein.sso.enabled=true
+gatein.sso.saml.sp.enabled=true
+gatein.sso.callback.enabled=true
+gatein.sso.valve.enabled=true
+gatein.sso.valve.class=org.gatein.sso.saml.plugin.valve.ServiceProviderAuthenticator
+gatein.sso.filter.login.sso.url=/portal/dologin
+gatein.sso.filter.initiatelogin.enabled=false
+gatein.sso.filter.logout.enabled=true
+gatein.sso.filter.logout.class=org.gatein.sso.saml.plugin.filter.SAML2LogoutFilter
+gatein.sso.filter.logout.url=${gatein.sso.sp.url}?GLO=true 
+gatein.sso.saml.config.file=${exo.conf.dir}/saml2/picketlink-sp.xml
+
+# Custom properties
+
+gatein.sso.sp.host=SP_HOSTNAME
+gatein.sso.sp.url=${gatein.sso.sp.host}/portal/dologin
+gatein.sso.idp.host=IDP_HOSTNAME
+gatein.sso.idp.url=IDP_SAML_ENDPOINT
+gatein.sso.idp.alias=IDP_SIGNING_ALIAS
+gatein.sso.idp.signingkeypass=IDP_SIGNING_KEY_PASS
+gatein.sso.idp.keystorepass=IDP_KEYSTORE_PASS
+# WARNING: This bundled keystore is only for testing purposes. You should generate and use your own keystore!
+gatein.sso.picketlink.keystore=${exo.conf.dir}/saml2/jbid_test_keystore.jks
+```
+
+You need to modify **gatein.sso.idp.host**, **gatein.sso.idp.url**, **gatein.sso.idp.alias**, **gatein.sso.idp.signingkeypass** and **gatein.sso.idp.keystorepass** according to your environment setup. You also need to install your own keystore as instructed in [Generating and using your own keystore](#generating-and-using-your-own-keystore).
+
+::: tip
+If your IDP send username in assertion with some char in capital letter, and you want to force lower case, you can add this property :
+
+```properties
+gatein.sso.saml.username.forcelowercase=true
+```
+
+3. Download and import your generated IDP certificate to your keystore
+   using this command:
+
+```bash
+keytool -import -keystore $PLATFORM_SP/gatein/conf/saml/jbid_test_keystore.jks -file idp-certificate.crt -alias Identity_Provider-idp
+```
+
+::: tip
+The Default password of the keystore jbid\_test\_keystore.jks is **store123**.
+
+4. Start up the platform: use the following command on Linux operating systems:
+```bash
+./start_eXo.sh
+```
+
+and use this command for Windows operating systems:
+```bash
+start_eXo.bat
+```
+
+
+
+## Generating and using your own keystore
+
+The default `jbid_test_keystore.jks` is useful for testing purpose,
+but in eXo Platform you need to generate and use your own keystore as
+follows:
+
+1. Generate your file using the **keytool** command:
+   
+```bash
+keytool -genkey -alias secure-key -keyalg RSA -keystore secure-keystore.jks
```

Comment:
Is there some requirements for how to generate the certificate and the keystore ? especially for the common name attribute ?

Reply:
I dont know exactly what are the requirement, and I think this part is not really our expertise, so I prefer to not add more details on that